### PR TITLE
remove onboarding background image, use gradient based on brand color

### LINKED ISSUE #19412: Remove background image as part of /onboarding config
## Is this feature related to a problem? Please describe.

As a Forem Admin, I want the background design of `/onboarding` to look clean and appealing without additional configuration.

In addition, as the creator of the Forem platform, we want the default configuration to be simplified so as to not overwhelm the Forem Admin with various configurations that can be burdensome to manage effectively.

## Describe the solution you’d like

Remove the background image option as part of `/onboarding`. This concerns removing it from the Admin > Customization > Config > Onboarding UI, removing the application logic and data, and also coming up with a new background design idea for `/onboarding`.

**This task should start as a design exploration to finalize the new look.**  We can then proceed into updating the UI, logic, and data layer.

**Design solution** : [Figma Designs](https://www.figma.com/file/u7VcNG4pjbn7K01V8JrFWo/Member-Onboarding-Process?type=design&node-id=2434-19415&t=boOnQK6YYswUr4Bm-4)

Replace the background image with branded gradient background. We have logic defined to find darker shades of branded color. For DEV we can use `indigo-600` as the brand color and `indigo-900` as the brand color - darkest shade.

Apart from that, we will have to remove the overlay background behind the modal. Below is the image dispalying how it would look for different Forems with different brand colors.

Final

## Definition of Done

**Task 1: Design exploration for new background look**

- [x] Anuj to explore new design ideas for the `/onboarding` background.
- [x] Take into account the "Brand Color" defined in Admin > Config > User Experience & Brand and incorporate it into the new design.

**Task 2: Update UI, logic, and data layer**

- [ ] Remove the “background image URL” field from Admin > Config > Onboarding
- [ ] Update the database schema to remove any tables or fields associated with the background image configuration. 

**Task 3: Implement the new background design**

- [ ] Based on the finalized design idea from Task 1, implement the new background design for `/onboarding`

## Additional Context

This is one of several issues collectively aimed at cleaning up the Onboarding to require fewer configurations and better defaults.

### PR DESCRIPTION
**Pull request by dukegreene**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

In an effort towards a cleaner and more consistent design with less configuration burden, we are simplifying from a settable background image for onboarding to an approach that draws a simpler color gradient based on the brand color from the instance's User Experience settings.

This PR adds a helper method to take in a hex (here, the brand color) and return an object with values for the light and dark gradient colors. Those color values are then passed from the data in the view render down to the onboarding component and used to build a gradient for the `background` value.

## Related Tickets & Documents

- Related Issue #19412
- Closes #19412

## QA Instructions, Screenshots, Recordings

One can visit /onboarding from any logged-in account, even after they have already completed onboarding. 

There should be no background image, and instead there should be a background gradient based on the instance's brand color.

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [X] Color contrast tested? *tl;dr the page background color will sometimes get close to the modal background color without sacrificing readability... Some values for darkened brand colors fail contrast checks against the black in the top 2/3 of the onboarding modal's background. But nothing in the modal background needs to be readable. The likely impact on accessibility is that some users will perceive the modal background as "melting into" the background of the page, but contrast ratios for readable text within the modal element will not be affected.*

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

pumbaa saying "oh the shade"

### CODE REVIEW

File: spec/helpers/color_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    it "accepts a hex string and returns an object with two colors for a gradient" do
+      yellow = "#fff000"
+      darker_yellow = "#9f9600"
+
+      expect(helper.gradient_from_hex(yellow)).to eq({ light: yellow, dark: darker_yellow })
+    end
+
+    it "fails gracefully when given a bad hex string" do
+      # default gradient is based on Dev.to brand defaults
+      expect(helper.gradient_from_hex("#oops")).to eq({ light: "#oops", dark: "#oops" })
```
Comment (rt4914) [inline spec/helpers/color_helper_spec.rb:14]: 🤣  ... oops

**Review by rt4914 — APPROVED** [review]
Clean PR. Great work @dukegreene

File: app/helpers/color_helper.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module ColorHelper
+  def gradient_from_hex(hex = "#4F46E5")
```
Comment (maestromac) [inline app/helpers/color_helper.rb:2]: `Settings::UserExperience` already has a default color of `#3b49df`. Is this fallback here necessary? I don't have a strong opinion.

**Review by maestromac — APPROVED** [review]
Tested locally, LGTM!

File: app/helpers/color_helper.rb (hunk 2 above)
Comment (dukegreene) [reply app/helpers/color_helper.rb:2]: Oh sure, this is an interesting piece that I should probably add a comment to. The gist of it is that the requested default values from the issue were indigo-600 and indigo-900 (based on darkenings of the default user experience color), and when I put the rgb of indigo-600 into a converter, #4f46e5 is what came out.

**Pull request merged**